wls-gui-wahllokalsystem/src/stores/stimmabgabevermerkeStore.ts

```diff
@@ -165,6 +165,36 @@ export const useStimmabgabevermerkeStore = defineStore(
       );
     }
 
+    function getBlattnummernThatPreventDeletion(newRowSize: number) {
+      const numbers: Set<number> = new Set<number>();
+      const currentLowestNumberOfRowsOverAllWahldaten =
+        lowestNumberOfRowsOverAllWahldaten.value;
+      if (newRowSize > 0 && currentLowestNumberOfRowsOverAllWahldaten) {
+        const removeRows =
+          newRowSize - currentLowestNumberOfRowsOverAllWahldaten - 1;
+        stimmabgabevermerke.value.forEach(
+          (stimmabgabevermerk: Stimmabgabevermerke) => {
+            // @ts-expect-error: noUncheckedIndexedAccess for wahldaten[0] | siehe #2008
+            const vermerke = stimmabgabevermerk.wahldaten[0].vermerke;
+            for (let i = removeRows; i < 0; i++) {
+              const vermerk = vermerke[vermerke.length + i];
+              if (
+                // @ts-expect-error: noUncheckedIndexedAccess for wahldaten[0] | siehe #2008
+                vermerk.stimmzettel.some(
+                  (stimmzettel) =>
+                    stimmzettel.anzahl != null && stimmzettel.anzahl != 0
+                )
+              ) {
+                // @ts-expect-error: noUncheckedIndexedAccess for wahldaten[0] | siehe #2008
+                numbers.add(vermerk.blattnummer);
+              }
```

Comment:
```suggestion
              if ( 
                vermerk &&
                vermerk.stimmzettel.some(
                  (stimmzettel) =>
                    stimmzettel.anzahl != null && stimmzettel.anzahl != 0
                )
              ) {
                numbers.add(vermerk.blattnummer);
              }
```

die ts errors haben nichts mehr mit dem wahdaten[0] zutun sondern kommen vom vermerk deswegen können wir die umgehen wenn wir ein `if (vermerk)` hinzufügen. oder alternativ die begründung im kommentar anpassen